src/clusterfuzz/_internal/bot/tasks/task_types.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Types of tasks. This needs to be seperate from commands.py because
+base/tasks.py depends on this module and many things commands.py imports depend
+on base/tasks.py (i.e. avoiding circular imports)."""
+from clusterfuzz._internal.bot.tasks import utasks
+from clusterfuzz._internal.metrics import logs
+
+
+class BaseTask:
+  """Base module for tasks."""
+
+  def __init__(self, module):
+    self.module = module
+
+  def execute(self, task_argument, job_type, uworker_env):
+    """Executes a task."""
+    raise NotImplementedError('Child class must implement.')
+
+
+class TrustedTask(BaseTask):
+  """Implementation of a task that is run on a single machine. These tasks were
+  the original ones in ClusterFuzz."""
+
+  def execute(self, task_argument, job_type, uworker_env):
+    # Simple tasks can just use the environment they don't need the uworker env.
+    del uworker_env
+    self.module.execute_task(task_argument, job_type)
+
+
+class UTask(BaseTask):
+  """Represents an untrusted task. Executes the preprocess part on this machine
```

Comment:
This is the final state we want to be in for most platforms right?

Reply:
Yes, it is the final state for all the tasks we plan to migrate. note that the only difference based on platform will be that things like mac bots will read from a queue to get their jobs instead of being scheduled on batch.